maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/StatelessXmlReporter.java

```diff
@@ -161,7 +161,7 @@ public void testSetCompleted(WrappedReportEntry testSetReportEntry, TestSetStats
                 OutputStreamWriter fw = getWriter(outputStream)) {
             XMLWriter ppw = new PrettyPrintXMLWriter(new PrintWriter(fw), XML_INDENT, XML_NL, UTF_8.name(), null);
 
-            createTestSuiteElement(ppw, testSetReportEntry, testSetStats); // TestSuite
+            createTestSuiteElement(ppw, testSetReportEntry, classMethodStatistics); // TestSuite
```

Comment:
Where this `TestSetStats` come from? Is it the listener event invoked by the test-provider?
Because, logically this `TestSetStats` makes sense.
Alright, we can accept this, we should consider this as a workaround because the real fix has to be in the Providers and it is not that easy and definitely it is not one PR.
I want to continue with the Roadmap in 3.0.0 milestones which solves multiple issues, and this one too, but it is so complex change that it requires multiple stable PRs promoted one by one.